lib/rubocop/cop/performance/zip_for_array_wrapping.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Performance
+      # Checks for `.map { |id| [id] }` and suggests replacing it with `.zip`.
+      #
+      # @example
+      #   # bad
+      #   [1, 2, 3].map { |id| [id] }
+      #
+      #   # bad
+      #   [1, 2, 3].map { [_1] }
+      #
+      #   # good
+      #   [1, 2, 3].zip
+      #
+      # @example
+      #   # good (no offense)
+      #   [1, 2, 3].map { |id| id }
+      #
+      # @example
+      #   # good (no offense)
+      #   [1, 2, 3].map { |id| [id, id] }
+      class ZipForArrayWrapping < Base
+        extend AutoCorrector
+
+        MSG = 'Use `zip` instead of `%<original_code>s`.'
+        RESTRICT_ON_SEND = Set.new(%i[map collect]).freeze
+
+        # @!method map_with_array?(node)
+        def_node_matcher :map_with_array?, <<~PATTERN
+          {
+            (block (send _ RESTRICT_ON_SEND) (args (arg _id)) (array (lvar _id)))
+            (numblock (send _ RESTRICT_ON_SEND) 1 (array (lvar _)))
+          }
+        PATTERN
+
+        def on_send(node)
+          return unless (parent = node.parent)
+          return unless map_with_array?(parent)
+          return unless (receiver = node.receiver&.source)
+
+          register_offense(parent, receiver, node)
+        end
+
+        private
+
+        def register_offense(parent, receiver, node)
+          add_offense(offense_range(parent), message: message(node)) do |corrector|
+            autocorrect(parent, receiver, corrector)
+          end
+        end
```

Comment:
```suggestion
        def register_offense(parent, node)
          offense_range = offense_range(parent)

          add_offense(offense_range(parent), message: message(node)) do |corrector|
            corrector.replace(offense_range, 'zip')
          end
        end
```

Since you're already getting the offense range, we can just replace it with `zip` without needing to replace the whole parent as well. A smaller surface area is preferable to avoid clobbering from other cops' corrections.

This suggestion would also make the `autocorrect` method unnecessary.

Reply:
Thanks, I was able to save another allocation after this as well